src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -865,6 +877,16 @@ public async Task UpdateLicenseAsync(Guid organizationId, OrganizationLicense li
                 }
             }
 
+            if (!license.UseKeyConnector && organization.UseKeyConnector)
+            {
+                var ssoConfig = await _ssoConfigRepository.GetByOrganizationIdAsync(organization.Id);
+                if (ssoConfig != null && ssoConfig.GetData().UseKeyConnector)
+                {
+                    throw new BadRequestException($"Your organization currently has Key Connector enabled. " +
+                        $"Your new license does not allow for the use of Key Connector. Disable your Key Connector.");
```

Comment:
Seems a bit cheeky telling them to disable something that they can't disable (yet), but it seems like an edge case that we can ignore for now.